idb/postgres/postgres.go

```diff
@@ -397,9 +397,29 @@ func (db *IndexerDb) AddBlock(block *bookkeeping.Block) error {
 // LoadGenesis is part of idb.IndexerDB
 func (db *IndexerDb) LoadGenesis(genesis bookkeeping.Genesis) error {
 	f := func(tx pgx.Tx) error {
+		// check networkdID
+		network, err := db.getMetastate(context.Background(), nil, schema.NetworkMetaStateKey)
+		if err == idb.ErrorNotInitialized {
+			networkState := types.NetworkState{
+				NetworkID: string(genesis.Network),
+			}
+			err = db.setMetastate(
+				nil, schema.NetworkMetaStateKey, string(encoding.EncodeNetworkState(&networkState)))
+			if err != nil {
+				return fmt.Errorf("LoadGenesis() err: %w", err)
+			}
+		} else {
+			networkState, err := encoding.DecodeNetworkState([]byte(network))
+			if err != nil {
+				return fmt.Errorf("LoadGenesis() err: %w", err)
+			}
+			if networkState.NetworkID != string(genesis.Network) {
+				return fmt.Errorf("LoadGenesis() err: NetworkID does not match")
```

Comment:
Styling: "err: " means `err` is a variable. Just write `return fmt.Errorf("LoadGenesis() NetworkID does not match")`.